test/NetTopologySuite.Samples.Console/Technique/SplitPolygonAtDateline.cs

```diff
@@ -0,0 +1,802 @@
+                             "(163 -14), (166.875 -11.8), (170 -10), (170 3.5))";
+            var rdr = new WKTReader();
+            var geom = rdr.Read(wktGeom);
+            var mps = (MultiPoint)geom;
+            var coords = mps.Coordinates.ToList();
+
+            var wtr = new WKTWriter();
+
+            // ========================================================================================
+            // function with defaults
+            // ========================================================================================
+            var resGeoms = ToPolyExOp(coords, wktPrj);
+            for (int i = 0; i < resGeoms.NumGeometries; i++)
+            {
+                var resGeom = resGeoms.GetGeometryN(i);
+                string resWkt = wtr.Write(resGeom);
+                Debug.Print("geometry {0} = {1}", i, resWkt);
+                DebugPrintCoords(resGeom.Coordinates.ToList(), "resGeom geom [" + i.ToString() + "] count");
+            }
+
+            // ========================================================================================
+            // function with specific parameter inputs. trim 0.75 on parts in the Eastern hemisphere and
+            // densify every 0.5 and attempt to fix invalid polygons.
+            // ========================================================================================
+            resGeoms = ToPolyExOp(coords, wktPrj, OgcGeometryType.Polygon, 0.75, 0.5, true);
+            for (int i = 0; i < resGeoms.NumGeometries; i++)
+            {
+                var resGeom = resGeoms.GetGeometryN(i);
+                string resWkt = wtr.Write(resGeom);
+                Debug.Print("geometry {0} = {1}", i, resWkt);
+                DebugPrintCoords(resGeom.Coordinates.ToList(), "resGeom geom [" + i.ToString() + "] count");
+            }
+
+            // ========================================================================================
+            // function with specific parameter inputs. trim 0.75 on parts in the Western hemisphere and
+            // densify every 0.25 and do not attempt to fix invalid polygons.
+            // ========================================================================================
+            resGeoms = ToPolyExOp(coords, wktPrj, OgcGeometryType.Polygon, -0.75, 0.25, false);
+            for (int i = 0; i < resGeoms.NumGeometries; i++)
+            {
+                var resGeom = resGeoms.GetGeometryN(i);
+                string resWkt = wtr.Write(resGeom);
+                Debug.Print("geometry {0} = {1}", i, resWkt);
+                DebugPrintCoords(resGeom.Coordinates.ToList(), "resGeom geom [" + i.ToString() + "] count");
+            }
+        }
+
+        /// <summary>
+        /// This routine takes a list of Coordinates and attempts to convert them to a poly geometry (polyline or polygon). Special care is
```

Comment:
What do I have to do when I have a `[Multi]LineString` that crosses the dateline?

Reply:
Right now I only implemented polygon. That's what I so very needed and I wanted to make sure that was solid before I started experimenting with polylines.